docker/docker_client.go

```diff
@@ -55,6 +58,7 @@ type bearerToken struct {
 // dockerClient is configuration for dealing with a single Docker registry.
 type dockerClient struct {
 	ctx             *types.SystemContext
+	httpWrapper     httpWrapper // nil unless running tests
```

Comment:
I still believe, in Golang, it's better to have an interface for this where it could be mocked in testing. This could be far better instead but far from what I think golang tests for this should be.